listenbrainz/mbid_mapping/mapping/dump.py

```diff
@@ -0,0 +1,219 @@
+            db_engine: an sqlalchemy Engine instance for making a connection
+            tables: a dict containing the names of the tables to be dumped as keys and the columns
+                    to be dumped as values
+            dump_time: the time at which the dump process was started
+
+        Returns:
+            the path to the archive file created
+    """
+
+    archive_name = 'metabrainz-metadata-dump-{time}'.format(
+        time=dump_time.strftime('%Y%m%d-%H%M%S')
+    )
+    archive_path = os.path.join(location, '{archive_name}.tar.zst'.format(
+        archive_name=archive_name,
+    ))
+
+    with open(archive_path, 'w') as archive:
+
+        zstd_command = ["zstd", "--compress", "-10"]
+        zstd = subprocess.Popen(zstd_command, stdin=subprocess.PIPE, stdout=archive)
+
+        with tarfile.open(fileobj=zstd.stdin, mode='w|') as tar:
+
+            temp_dir = tempfile.mkdtemp()
+
+            try:
+                timestamp_path = os.path.join(temp_dir, "TIMESTAMP")
+                with open(timestamp_path, "w") as f:
+                    f.write(dump_time.isoformat(" "))
+                tar.add(timestamp_path,
+                        arcname=os.path.join(archive_name, "TIMESTAMP"))
+                tar.add(DUMP_LICENSE_FILE_PATH,
+                        arcname=os.path.join(archive_name, "COPYING"))
+            except IOError as e:
+                current_app.logger.error(
+                    'IOError while adding dump metadata: %s', str(e), exc_info=True)
+                raise
+            except Exception as e:
+                current_app.logger.error(
+                    'Exception while adding dump metadata: %s', str(e), exc_info=True)
+                raise
+
+            archive_tables_dir = os.path.join(temp_dir, 'metabrainz')
+            create_path(archive_tables_dir)
+
+            with db_engine.connect() as connection:
+                with connection.begin() as transaction:
+                    cursor = connection.connection.cursor()
+                    for table in tables:
+                        try:
+                            copy_table(
+                                cursor=cursor,
+                                location=archive_tables_dir,
+                                columns=tables[table]['columns'],
+                                table_name=table,
+                            )
+                        except IOError as e:
+                            current_app.logger.error(
+                                'IOError while copying table %s', table, exc_info=True)
+                            raise
+                        except Exception as e:
+                            current_app.logger.error(
+                                'Error while copying table %s: %s', table, str(e), exc_info=True)
+                            raise
+                    transaction.rollback()
+
+            # Add the files to the archive in the order that they are defined in the dump definition.
```

Comment:
Since no FKs or import concern apply here, probably best to simplify the comment to just `Add the files to the archive in the order that they are defined in the dump definition.`